src/schedule/schedule_strategy.py

```diff
@@ -20,10 +31,11 @@ def from_schedule_configuration(cls, schedule_configuration: ScheduleConfigurati
         """
         raise NotImplementedError()
 
-    @abstractmethod
     def should_spawn(self, tick: int) -> bool:
```

Comment:
Maybe this method should be renamed to 'is_between_start_and_end_tick' and 'should_spawn' should stay abstract. Do you think, that this will ever be the only check for any strategy to determine if it should spawn?

Reply:
The inheriting classes also implement `should_spawn` and then call `super().should_spawn`. So every class adds its own logic and implements it at its own inheritance level.

Reply:
If I have to add another check I simply add it the `should_spawn` of the responsible class.